src/v/kafka/server/group_manager.cc

```diff
@@ -86,6 +101,57 @@ void group_manager::attach_partition(ss::lw_shared_ptr<cluster::partition> p) {
     _partitions.rehash(0);
 }
 
+ss::future<> group_manager::cleanup_removed_topic_partitions(
+  const std::vector<model::topic_partition>& tps) {
+    using group_info = absl::node_hash_map<group_id, group_ptr>::value_type;
+    return ss::do_for_each(_groups, [this, &tps](group_info& gi) {
+        return gi.second->remove_topic_partitions(tps).then(
+          [this, g = gi.second] {
+              if (!g->in_state(group_state::dead)) {
+                  return ss::now();
+              }
+              auto it = _groups.find(g->id());
+              if (it == _groups.end()) {
+                  return ss::now();
+              }
+              // ensure the group didn't change
+              if (it->second != g) {
+                  return ss::now();
+              }
+              vlog(klog.trace, "Removed group {}", g);
+              _groups.erase(it);
+              _groups.rehash(0);
+              return ss::now();
+          });
+    });
+}
+
+void group_manager::handle_topic_delta(
+  const std::vector<cluster::topic_table_delta>& deltas) {
+    // topic-partition deletions in the kafka namespace are the only deltas that
+    // are relevant to the group manager
+    std::vector<model::topic_partition> tps;
+    for (const auto& delta : deltas) {
+        if (
+          delta.type == cluster::topic_table_delta::op_type::del
+          && delta.ntp.ns == model::kafka_namespace) {
+            tps.emplace_back(delta.ntp.tp);
+        }
+    }
+
+    if (tps.empty()) {
+        return;
+    }
+
+    (void)ss::with_gate(_gate, [this, tps = std::move(tps)]() mutable {
+        return ss::do_with(
+          std::move(tps),
+          [this](const std::vector<model::topic_partition>& tps) {
+              return cleanup_removed_topic_partitions(tps);
+          });
+    });
```

Comment:
we should catch this gate exception and decide what to do w/ it. If logging is OK or explicitly crash. Either way we should handle this background exception of the gate.

Reply:
yes, good call. oops